mask/eolearn/mask/cloud_mask.py

```diff
@@ -292,3 +297,592 @@ def get_s2_pixel_cloud_detector(threshold=0.4, average_over=4, dilation_size=2,
+        Add selected features (cloud probabilities and masks) to an EOPatch instance.
+
+        :param eopatch: Input `EOPatch` instance
+        :return: `EOPatch` with additional features
+        """
+
+        # Get data and is_data
+        feature_type, feature_name = next(self.data_feature(eopatch))
+        bands = eopatch[feature_type][feature_name][..., self.band_indices].astype(np.float32)
+
+        feature_type, feature_name = next(self.is_data_feature(eopatch))
+        is_data = eopatch[feature_type][feature_name].astype(bool)
+
+        original_shape = bands.shape[1:-1]
+        scale_factors, self.sigma = self._parse_resolution_data(original_shape, eopatch.meta_info)
+
+        mono_proba_feature, mono_mask_feature = self.mono_features
+        multi_proba_feature, multi_mask_feature = self.multi_features
+
+        # Downscale if specified
+        if scale_factors is not None:
+            bands = resize_images(bands.astype(np.float32), scale_factors=scale_factors)
+            is_data_sm = resize_images(is_data.astype(np.uint8), scale_factors=scale_factors).astype(np.bool)
+
+
+        mono_proba = None
+        multi_proba = None
+
+        # Run s2cloudless if needed
+        if any(feature is not None for feature in [self.mask_feature, mono_mask_feature, mono_proba_feature]):
+            mono_proba = self._mono_iterations(bands)
+            mono_proba = mono_proba.reshape(*bands.shape[:-1], 1)
+
+            # Upscale if necessary
+            if scale_factors is not None:
+                mono_proba = resize_images(mono_proba, new_size=original_shape)
+
+            # Average over and threshold
+            mono_mask = self._average_all(mono_proba) >= self.mono_threshold
+
+        # Run SSIM-based multi-temporal classifier if needed
+        if any(feature is not None for feature in [self.mask_feature, multi_mask_feature, multi_proba_feature]):
+            multi_proba = self._multi_iterations(bands, is_data_sm)
+            multi_proba = multi_proba.reshape(*bands.shape[:-1], 1)
+
+            # Upscale if necessary
+            if scale_factors is not None:
+                multi_proba = resize_images(multi_proba, new_size=original_shape)
+
+            # Average over and threshold
+            multi_mask = self._average_all(multi_proba) >= self.multi_threshold
+
+
+        # Intersect
+        if self.mask_feature is not None:
+            inter_mask = mono_mask & multi_mask
+
+        if mono_mask_feature is not None:
+            mono_mask = self._dilate_all(mono_mask)
+            eopatch.mask[mono_mask_feature] = (mono_mask * is_data).astype(bool)
+
+        if multi_mask_feature is not None:
+            multi_mask = self._dilate_all(multi_mask)
+            eopatch.mask[multi_mask_feature] = (multi_mask * is_data).astype(bool)
+
+        if self.mask_feature is not None:
+            inter_mask = self._dilate_all(inter_mask)
+            eopatch.mask[self.mask_feature] = (inter_mask * is_data).astype(bool)
```

Comment:
Same as before for `mono_mask`, `multi_mask` and `inter_mask`

Reply:
If you look at the code before, when at least one of `self.mask_feature, mono_mask_feature, mono_proba_feature` is not None `mono_mask` is computed. Similar for `multi_mask`.

Therefore the when the mentioned if statements will be true (and executed), the required masks will always be computed.

Reply:
OK, no problem. I just wanted to point this out as a possible problem, without really going deep into it. Thanks!